.github/workflows/CI.yml

```diff
@@ -45,9 +45,9 @@ jobs:
       with:
         java-version: ${{ matrix.java }}
         distribution: temurin
-    - name: Build and Run Tests
+    - name: Build
       run: |
-          ./gradlew check
+          ./gradlew check -x test -x integTest -x yamlRestTest
     - name: Upload Coverage Report
       if: matrix.os == 'ubuntu-latest'
       uses: codecov/codecov-action@v3
```

Comment:
Actually... codecov may depend on the tests being run. Does this still work?

Reply:
It didn't run on this PR. That's something to think about. Do you think I should try moving codecov to tests.yml?

Reply:
Maybe. I'm still not clear on the motivation to split everything out from "check" which did it all anyway.

Reply:
Also since each step in the tests.yml does its own checkout, I'm not sure if the coverage would be cumulative for all the steps.

Reply:
Yaay! Codecov worked!

Reply:
The motivation behind is to run the tests and build in parallel and reduce the CI time overall when our plugin will become more heavy.

Reply:
> run the tests and build in parallel

Hmm.  

Full gradle check gives a total time of 50.570s.  
Gradle check without the tests takes 9.819s.
Running the tests separately gives 42.721s, because you're still running them sequentially.

You could get them to run in parallel by putting them up under the highest level `jobs:` label:

| Task | Duration |
| --- | --- |
| :integTest | 14.761s |  
| :test | 14.051s |  
| :yamlRestTest | 10.518s

Note jacoco is only considering unit tests, so you could (for now) put jacoco in that job and run the other two jobs without jacoco.
```groovy
jacocoTestReport {
    dependsOn test
    reports {
        xml.required = true
    }
}
```
There are ways to get jacoco to combine others: https://docs.gradle.org/7.4-rc-1/userguide/test_report_aggregation_plugin.html but I'm not sure that's needed.